File: 351004/Romanovskiy/task310/src/main/java/com/example/task310/mapper/TweetMapper.java
package com.example.task310.mapper;

import com.example.task310.domain.dto.request.TweetRequestTo;
import com.example.task310.domain.dto.response.TweetResponseTo;
import com.example.task310.domain.entity.Tweet;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface TweetMapper {
    Tweet toEntity(TweetRequestTo request);

    TweetResponseTo toResponse(Tweet tweet);
}

File: 351002/Abrashin/src/main/kotlin/com/example/distcomp/mapper/TweetMapper.kt
package com.example.distcomp.mapper

import com.example.distcomp.dto.request.TweetRequestTo
import com.example.distcomp.dto.response.TweetResponseTo
import com.example.distcomp.model.Tweet
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface TweetMapper {
    fun toEntity(request: TweetRequestTo): Tweet
    fun toResponse(entity: Tweet): TweetResponseTo
}
